### ISSUE TITLE
Support for an osb init tool to allow for different project structure.

### ISSUE THREAD

**Issue by maleck13** [openshift/open-service-broker-sdk#17]
Wondering would some form of open-service-broker tool, similar to the cobra cli tool be valuable here? 
Allowing for something like:
```
osb init

```
That would scaffold the users OSB project based on the structure of the sdk project ensuring the import paths were correct. I guess using the open-service-broker-sdk as a template.
The reason for this is that when setting up my environment to use the sdk, I wanted to have it in my own project structure:
```
github.com/maleck13/mongo-service-broker

```
But of course this causes issues as everything is imported from:
```
github.com/openshift/open-service-broker-sdk

```

Switching these import paths solved the problem but there were many of them to change.

**Comment by bparees** [comment]
How about we just provide a sed script?  it's a pretty trivial global search/replace, I think...

**Comment by maleck13** [comment]
Seems reasonable. A tool is probably overkill unless other requirements emerged

Craig

**Comment by maleck13** [comment]
@bparees Thank you for all your help. It is appreciated. As an additional note on things I had to change. I also made changes in the templates to change the name from brokerdsk to my app name.

I also have a question about setup, if you want me to create a new issue around this, just let me know. 

I have a  Kubernetes 1.6.0 cluster with service catalog running on it. I went through their walkthrough and their UPS broker worked ok.
After forking this repo and starting my own broker, I attempted to install it on my Kubernetes cluster. I got the broker up and running, but the catalog failed to retrieve the service classes. Digging in to this I discovered t was down to authorisation.
The rules in test-scripts/resources/roles.yaml seem like they are there to solve this issue? Unfortunately they would not create on my Kubernetes cluster. 
```
couldn't find type: v1.ClusterRole
```
I assume these are OpenShift specific? 
I tried switching them over to 
```
rbac.authorization.k8s.io/v1beta1
```
But some things are different and I don't have enough of an understanding of the rbac setup in Kube to solve them (yet).

So 3 questions
1) Is it possible to install service catalog on the latest ```oc cluster up / minishift``` ? as this would get me past this.
2) Are you familiar enough with rbac to translate the roles.yaml?
3) Is it the aim of this project to support both Kubernetes and OpenShift setups?

**Comment by bparees** [comment]
> The rules in test-scripts/resources/roles.yaml seem like they are there to solve this issue?

yes for now those scripts open the api so it can be invoked w/o auth

> I assume these are OpenShift specific?

yes, sounds like we need to create some k8s versions of the setup resources+install script.

> Is it possible to install service catalog on the latest oc cluster up / minishift ? as this would get me past this.

we're working on adding the service catalog to cluster up this sprint but there are some prereqs that need to be met before we can do so.

> Are you familiar enough with rbac to translate the roles.yaml?

no but i can probably find someone who is.  Can you open an issue for this so we can loop the right people in?

> Is it the aim of this project to support both Kubernetes and OpenShift setups?

yes.
